recipes/server_extend.rb

```diff
@@ -16,30 +16,37 @@
       Chef::Log.warn("Unable to find a chef node for #{peer}")
       next
     end
-    peer_bricks = chef_node['gluster']['server']['bricks'].select { |brick| brick.include? volume_name }
+
+    unless node.default['gluster']['server']['volumes'][volume_name].attribute?('bricks_waiting_to_join')
+      node.default['gluster']['server']['volumes'][volume_name]['bricks_waiting_to_join'] = ''
+    end
+
+    peer_bricks = chef_node['gluster']['server']['volumes'][volume_name]['bricks'].select { |brick| brick.include? volume_name }
     brick_count += (peer_bricks.count || 0)
     peer_bricks.each do |brick|
       Chef::Log.info("Checking #{peer}:#{brick}")
       unless brick_in_volume?(peer, brick, volume_name)
-        node.default['gluster']['server']['bricks_waiting_to_join'] << " #{peer}:#{brick}"
+        node.default['gluster']['server']['volumes'][volume_name]['bricks_waiting_to_join'] << " #{peer}:#{brick}"
       end
     end
   end
 
   replica_count = volume_values['replica_count']
-  next if node['gluster']['server']['bricks_waiting_to_join'].empty?
+  next if node['gluster']['server']['volumes'][volume_name]['bricks_waiting_to_join'].empty?
   # The number of bricks in bricks_waiting_to_join has to be a modulus of the replica_count we are using for our gluster volume
   if (brick_count % replica_count) == 0
     Chef::Log.info("Attempting to add new bricks into volume #{volume_name}")
-    execute "gluster volume add-brick #{volume_name} #{node['gluster']['server']['bricks_waiting_to_join']}" do
+    execute "gluster volume add-brick #{volume_name} #{node['gluster']['server']['volumes'][volume_name]['bricks_waiting_to_join']}" do
       action :run
     end
+    node.set['gluster']['server']['volumes'][volume_name]['bricks_waiting_to_join'] = ''
   elsif volume_values['volume_type'] == 'replicated' || volume_values['volume_type'] == 'striped'
     Chef::Log.warn("#{volume_name} is a replicated or striped volume, adjusting replica count to match new number of bricks")
     node.set['gluster']['server']['volumes'][volume_name][replica_count] = brick_count
-    execute "gluster volume add-brick replica #{brick_count} #{volume_name} #{node['gluster']['server']['bricks_waiting_to_join']}" do
+    execute "gluster volume add-brick replica #{volume_name} #{brick_count} #{node['gluster']['server']['volumes'][volume_name]['bricks_waiting_to_join']}" do
```

Comment:
also, in the `elsif` statement, you allow `striped`, but have `replica` hardcoded in the add-brick command